zcash_proofs/src/circuit/sapling.rs

```diff
@@ -28,10 +28,26 @@ use group::ff::PrimeFieldBits;
 
 pub const TREE_DEPTH: usize = SAPLING_COMMITMENT_TREE_DEPTH;
 
+/// The preimage of a Sapling value commitment.
+#[derive(Clone)]
+pub struct ValueCommitmentPreimage {
```

Comment:
Consider renaming this to `ValueCommitmentOpening`. (I was initially confused what this was when calling it a preimage. In particular, "opening" unambiguously includes the randomness.)